tests/enlyze/test_validators.py

```diff
@@ -0,0 +1,80 @@
+from contextlib import nullcontext as does_not_raise
+from datetime import datetime, timedelta
+from uuid import UUID
+
+import pytest
+from hypothesis import given
+from hypothesis import strategies as st
+
+from enlyze.errors import EnlyzeError, ResamplingValidationError
+from enlyze.models import Variable, VariableWithResamplingMethod
+from enlyze.validators import (
+    validate_resampling_interval,
+    validate_timeseries_arguments,
+)
+from tests.conftest import (
+    datetime_before_today_strategy,
+    datetime_today_until_now_strategy,
+)
+
+variable_strategy = st.one_of(
+    st.builds(Variable), st.builds(VariableWithResamplingMethod)
+)
+
+
+class TestValidateTimeseriesArguments:
+    @given(
+        start=datetime_before_today_strategy,
+        end=datetime_today_until_now_strategy,
+        variable=variable_strategy,
+    )
+    def test_validate_timeseries_arguments(self, start, end, variable):
+        start, end, appliance_uuid = validate_timeseries_arguments(
+            start, end, [variable]
+        )
+        assert start
+        assert end
+        assert UUID(appliance_uuid)
+
+    @given(variable=variable_strategy)
+    def test_validate_start_must_be_earlier_than_end(self, variable):
+        start = datetime.now() + timedelta(days=1)
+        end = datetime.now()
```

Comment:
Calling `now()` twice is not good. 

```suggestion
        end = datetime.now()
        start = start + timedelta(days=1)
```

Reply:
Why though?

Reply:
It literally doesn't matter at all in our case, anyways, incorporated.